patches/tModLoader/Terraria/Recipe.Extensions.cs

```diff
@@ -47,17 +37,7 @@ public partial class Recipe
 
 		public bool HasIngredient<T>() where T : ModItem => HasIngredient(ModContent.ItemType<T>());
 
-		public bool HasRecipeGroup(int id) {
-			foreach (int groupID in acceptedGroups) {
-				if (groupID == -1) break;
-
-				if (groupID == id) {
-					return true;
-				}
-			}
-
-			return false;
-		}
+		public bool HasRecipeGroup(int id) => acceptedGroups.Any(group => group == id);
```

Comment:
`Contains`. Please

Reply:
Woops, will fix in the main branch once this PR gets merged

Reply:
Just fixed this.